functions/src/cron-jobs/specialists.ts

```diff
@@ -0,0 +1,111 @@
+      })
+      .eachPage(
+        function page(records: any, fetchNextPage: any) {
+          allSkills.push(
+            ...records.map((record: any) => {
+              const intermediate = record.get(SKILL_FIELDS.INTERMEDIATE) || [];
+              const experts = record.get(SKILL_FIELDS.EXPERT) || [];
+              const specialistsAmount = experts.length + intermediate.length;
+
+              const skill: SkillRecord = {
+                label: record.get(SKILL_FIELDS.NAME),
+                specialistsAmount,
+              };
+              return skill;
+            })
+          );
+
+          fetchNextPage();
+        },
+        function done(err: any) {
+          if (err) {
+            functions.logger.error(`Failed to get Airtable data: ${err}`);
+            return reject({});
+          }
+          const skillsReduced = allSkills.reduce((result, skill) => {
+            result[skill.label] = skill.specialistsAmount;
+            return result;
+          }, {} as Record<string, number>);
+
+          return resolve(skillsReduced);
+        }
+      );
+  });
+};
+
+const getContentfulData = async (): Promise<Entry[]> => {
+  return new Promise(async (resolve, reject) => {
+    const { CONTENTFUL_TOKEN, CONTENTFUL_SPACE_ID, CONTENTFUL_ENVIRONMENT } = getSecrets();
+    try {
+      const client = getContentfulClient(CONTENTFUL_TOKEN);
+      const contentfulConfig = { space: CONTENTFUL_SPACE_ID, environment: CONTENTFUL_ENVIRONMENT };
+      const environment = await getEnvironment(client, contentfulConfig);
+      const entries = await environment.getEntries({ content_type: 'entry', limit: 700 });
+      resolve(entries.items);
+    } catch (err) {
+      functions.logger.error(`Failed to get Contentful data: ${err}`);
+      reject([]);
+    }
+  });
+};
+
+export const updateSpecialistsAmount = functions
+  .runWith({ secrets: [...AIRTABLE_SECRET_KEYS, ...CONTENTFUL_SECRET_KEYS, ...URL_SECRET_KEYS] })
+  .pubsub.schedule('0 9 1,15 * *') // runs at 1st and 15th day of every month 9:00
+  .timeZone(DEFAULT_TIMEZONE)
+  .onRun(async () => {
+    const allSkills = await getAirtableData();
+    const contentfulEntries = await getContentfulData();
+    const entries = contentfulEntries.map((entry) => {
+      const label = entry.fields.label?.['en-US'];
+      const currentAmount = entry.fields.experts?.['en-US'];
+      return {
+        label,
+        newAmount: allSkills[label],
+        currentAmount: Number(currentAmount ?? 0),
+        entry,
+      };
+    });
+    entries.forEach((technology, i) => {
+      setTimeout(() => {
+        const { label, currentAmount, newAmount, entry } = technology;
+        if (!newAmount || currentAmount === newAmount) return;
+        try {
+          functions.logger.info(`Changing ${label} experts field from ${currentAmount} to ${newAmount}`);
+          entry.fields.experts = { 'en-US': newAmount.toString() };
```

Comment:
You're partially right, the model was wrong because it was made to be a `string` before. I've changed it into a number, but it still needs locale-specific value. This is a screenshot of `getEntries` method, as you can see even `moved` property has `en-US` field despite being a number

<img width="345" alt="image" src="https://github.com/apptension/tech-radar/assets/53352540/5c15c908-ba46-4e77-923c-d914a49aeb21">